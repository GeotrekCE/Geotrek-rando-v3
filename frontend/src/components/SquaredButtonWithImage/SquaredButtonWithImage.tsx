import Image from 'next/image';
import { FormattedMessage } from 'react-intl';

interface SquaredButtonWithImageProps {
  titleKey: string;
  imageUrl: string;
}

export const SquaredButtonWithImage: React.FC<SquaredButtonWithImageProps> = ({
  imageUrl,
  titleKey,
}) => {
  return (
    <div className="h-12 w-12 relative overflow-hidden shadow-md rounded-medium border-2 border-solid border-white">
      <Image
        loading="lazy"
        src={imageUrl}
        className="object-cover h-full w-full rounded-medium"
        width={44}
        height={44}
        alt=""
      />
      <span
        className="font-bold text-white text-CTA text-center w-full
      absolute bottom-1 left-0 right-0 z-text"
      >
        <FormattedMessage id={titleKey} />
      </span>
    </div>
  );
};
