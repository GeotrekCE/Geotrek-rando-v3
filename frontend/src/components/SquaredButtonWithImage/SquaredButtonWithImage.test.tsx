import React from 'react';
import { render } from 'services/testing/reactTestingLibraryWrapper';
import { SquaredButtonWithImage } from './SquaredButtonWithImage';

describe('SquaredButtonWithImage', () => {
  it('renders correctly', () => {
    const squaredButtonWithImage = render(
      <SquaredButtonWithImage imageUrl="imageUrl.jpg" titleKey="map.layerButton.satellite" />,
    );

    expect(squaredButtonWithImage).toMatchSnapshot();
  });
});
