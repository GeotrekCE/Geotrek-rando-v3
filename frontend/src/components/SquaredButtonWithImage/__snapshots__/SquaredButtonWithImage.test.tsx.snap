// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquaredButtonWithImage renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-12 w-12 relative overflow-hidden shadow-md rounded-medium border-2 border-solid border-white"
      >
        <img
          alt=""
          class="object-cover h-full w-full rounded-medium"
          data-nimg="1"
          decoding="async"
          height="44"
          loading="lazy"
          src="/_next/image?url=%2FimageUrl.jpg&w=96&q=75"
          srcset="/_next/image?url=%2FimageUrl.jpg&w=48&q=75 1x, /_next/image?url=%2FimageUrl.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="44"
        />
        <span
          class="font-bold text-white text-CTA text-center w-full absolute bottom-1 left-0 right-0 z-text"
        >
          Satellite
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-12 w-12 relative overflow-hidden shadow-md rounded-medium border-2 border-solid border-white"
    >
      <img
        alt=""
        class="object-cover h-full w-full rounded-medium"
        data-nimg="1"
        decoding="async"
        height="44"
        loading="lazy"
        src="/_next/image?url=%2FimageUrl.jpg&w=96&q=75"
        srcset="/_next/image?url=%2FimageUrl.jpg&w=48&q=75 1x, /_next/image?url=%2FimageUrl.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="44"
      />
      <span
        class="font-bold text-white text-CTA text-center w-full absolute bottom-1 left-0 right-0 z-text"
      >
        Satellite
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
